src/plugins/hetero/src/plugin.cpp

```diff
@@ -0,0 +1,289 @@
+        return rw_properties;
+    };
+    const auto& to_string_vector = [](const std::vector<ov::PropertyName>& properties) {
+        std::vector<std::string> ret;
+        for (const auto& property : properties) {
+            ret.emplace_back(property);
+        }
+        return ret;
+    };
+    if (METRIC_KEY(SUPPORTED_METRICS) == name) {
+        auto metrics = default_ro_properties();
+
+        add_ro_properties(METRIC_KEY(SUPPORTED_METRICS), metrics);
+        add_ro_properties(METRIC_KEY(SUPPORTED_CONFIG_KEYS), metrics);
+        add_ro_properties(METRIC_KEY(IMPORT_EXPORT_SUPPORT), metrics);
+        return to_string_vector(metrics);
+        // IE_SET_METRIC_RETURN(SUPPORTED_METRICS,
+        //                     // TODO: check list
+        //                      std::vector<std::string>{METRIC_KEY(SUPPORTED_METRICS),
+        //                                               ov::device::full_name.name(),
+        //                                               METRIC_KEY(SUPPORTED_CONFIG_KEYS),
+        //                                               METRIC_KEY(IMPORT_EXPORT_SUPPORT),
+        //                                               ov::caching_properties.name(),
+        //                                               ov::device::capabilities.name()});
+    } else if (METRIC_KEY(SUPPORTED_CONFIG_KEYS) == name) {
+        return to_string_vector(default_rw_properties());
+        // IE_SET_METRIC_RETURN(SUPPORTED_CONFIG_KEYS, getSupportedConfigKeys());
+    } else if (ov::supported_properties == name) {
+        auto ro_properties = default_ro_properties();
+        auto rw_properties = default_rw_properties();
+
+        std::vector<ov::PropertyName> supported_properties;
+        supported_properties.reserve(ro_properties.size() + rw_properties.size());
+        supported_properties.insert(supported_properties.end(), ro_properties.begin(), ro_properties.end());
+        supported_properties.insert(supported_properties.end(), rw_properties.begin(), rw_properties.end());
+        return decltype(ov::supported_properties)::value_type(supported_properties);
+    } else if (ov::device::full_name == name) {
+        return decltype(ov::device::full_name)::value_type{"HETERO"};
+    } else if (METRIC_KEY(IMPORT_EXPORT_SUPPORT) == name) {
+        return true;
+    } else if (ov::caching_properties == name) {
+        // TODO vurusovs: RECHECK WITH ov::hetero::caching_device_properties
+        return decltype(ov::caching_properties)::value_type{ov::hetero::caching_device_properties.name()};
+    } else if (ov::hetero::caching_device_properties == name) {
+        // std::string targetFallback = GetTargetFallback(user_options);
+        // it = hetero_config.find(ov::device::priorities.name());
+
+        // TODO vurusovs: CHECK `target_fallback` is empty or not
+        // TODO vurusovs: RECHECK WITH ov::caching_properties
+        auto target_fallback = m_cfg.device_priorities;
```

Comment:
Why you ignore `arguments`?

Reply:
It was done by mistake, fixed, thank you!